external-services/docker-compose.yml

```diff
@@ -1,31 +1,31 @@
 version: "3.9"
 services:
-  hash-dev-opensearch:
-    deploy:
-      restart_policy:
-        condition: on-failure
-    environment:
-      ## Tell OpenSearch that it's operating in single-node mode
-      - discovery.type=single-node
-      ## Disable the security module for development so we can connect over plain HTTP
-      - plugins.security.disabled=true
-      ## Docker volumes are ~10GB by default which is typically much smaller than the
-      ## host's drive size. This can cause OpenSearch to shutdown if it thinks disk
-      ## space is running low. Set the disk high watermark to 100% to ignore this.
-      - cluster.routing.allocation.disk.watermark.high=100%
-      - cluster.routing.allocation.disk.watermark.flood_stage=100%
-    image: opensearchproject/opensearch:1.2.3
-    ports:
-      - 9200:9200
-    ulimits:
-      nofile:
-        soft: 65536
-        hard: 65536
-    ## Mouting open search data to a local directory may lead to java.nio.file.AccessDeniedException.
-    ## Details: https://github.com/opensearch-project/OpenSearch/issues/1579.
-    ## We can revisit the setup after upgrading base image or by fixing permissions in a custom image.
-    # volumes:
-    #   - ../var/external-services/opensearch/data:/usr/share/opensearch/data
+  # hash-dev-opensearch:
+  #   deploy:
+  #     restart_policy:
+  #       condition: on-failure
+  #   environment:
+  #     ## Tell OpenSearch that it's operating in single-node mode
+  #     - discovery.type=single-node
+  #     ## Disable the security module for development so we can connect over plain HTTP
+  #     - plugins.security.disabled=true
+  #     ## Docker volumes are ~10GB by default which is typically much smaller than the
+  #     ## host's drive size. This can cause OpenSearch to shutdown if it thinks disk
+  #     ## space is running low. Set the disk high watermark to 100% to ignore this.
+  #     - cluster.routing.allocation.disk.watermark.high=100%
+  #     - cluster.routing.allocation.disk.watermark.flood_stage=100%
+  #   image: opensearchproject/opensearch:1.2.3
+  #   ports:
+  #     - 9200:9200
+  #   ulimits:
+  #     nofile:
+  #       soft: 65536
+  #       hard: 65536
+  #   ## Mouting open search data to a local directory may lead to java.nio.file.AccessDeniedException.
+  #   ## Details: https://github.com/opensearch-project/OpenSearch/issues/1579.
+  #   ## We can revisit the setup after upgrading base image or by fixing permissions in a custom image.
+  #   # volumes:
+  #   #   - ../var/external-services/opensearch/data:/usr/share/opensearch/data
```

Comment:
Let me try to play around with this a little. Maybe there is a cheap way to create a custom entry point that will enter "noop" mode based on some env var.